src/core/server/http_resources/get_apm_config.ts

```diff
@@ -6,36 +6,41 @@
  * Side Public License, v 1.
  */
 
-import { getConfig } from '../../../apm';
 import agent from 'elastic-apm-node';
+// @ts-expect-error apm module is a js file outside of core (need to split APM/rum configuration)
+import { getConfig } from '../../../apm';
```

Comment:
Currently, the RUM and server-side apm configurations are shared, and the logic to load it is defined in `src/apm.js`.

As for the server-side, the apm configuration needs to be loaded before everything, I don't think we can easily move `/src/apm.js` to `core` (or even convert it to TS).

If that's ok with everyone, I will open a follow-up discussion issue to see our options here. As stated in the comment I added, we may want to split the apm and the RUM configuration loading logic.

Reply:
Atm the `apm.js` file is imported extremely early in the entrypoint. AFAIK, that's even the first thing that is imported, even before we setup node env:

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/cli/dev.js#L9-L12

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/cli/dist.js#L9-L11

This is to be sure to have the module interceptor online before anything else (even before we register the TS->JS transpilation in `setup_node_env`)

- Converting to TS would force to change the `dev` and `dist` load order, which would probably have impact on the modules loaded before the agent is online

- Moving it to core would still force to use deep import from these `dev` and `dist` script, as an import such as `import { getApmConfig} from '../src/core'` would load core and all underlying imports before the apm agent.

Reply:
that makes sense. Can we refactor `src/apm.js` to a package?

Reply:
Would need to double check, but I think we do. May even solve the typescript issue as our packages are pre-transpiled to JS

Reply:
Created https://github.com/elastic/kibana/issues/93823